### ISSUE TITLE
System error on opening/switching/closing projects

### ISSUE THREAD

**Issue by krauthex** [rstudio/rstudio#15318]
### System details

    RStudio Edition : Server
    RStudio Version :  2024.09.0-375
    OS Version      :  Debian 12 (bookworm)
    R Version       :  4.2.1 (2022-06-23)

### Steps to reproduce the problem
1. Start new Session or open/switch/close a project
2. error message pops up:
![image](https://github.com/user-attachments/assets/58f34e9a-fa10-4146-8869-c450d0c9d0fc)

This happens 2-3 times, then suddenly it works again. Sometimes this also happens directly after logging in to Rstudio-server when the server is trying to restore a project.

#### VM setup
Access to the VM is configured via ldap, so users can either `ssh` to the VM or log into rstudio using their ldap username+password. Additionally, the home directories are located on (global) NFS shares. The issue also existed on previous RStudio-server versions.

#### rstudio-server logs
The logs of the `rstudio-server.service` show the following (multiple) error messages:
```console
Oct 15 09:07:13 vsci01-dev rsession-USER[260648]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:07:13 vsci01-dev rsession-USER[260648]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:07:13 vsci01-dev rsession-USER[260648]: WARNING Skipping source db move from: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/per/t/A509CED8-contents to /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/A509CED8-contents; LOGGED FROM: void rstudio::session::source_database::supervisor::{anonymous}::attemptToMoveSourceDbFiles(const rstudio::core::FilePath&, const rstudio::core::FilePath&) src/cpp/session/SessionSourceDatabaseSupervisor.cpp:215
Oct 15 09:07:13 vsci01-dev rsession-USER[260648]: WARNING Skipping source db move from: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/per/t/A509CED8 to /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/A509CED8; LOGGED FROM: void rstudio::session::source_database::supervisor::{anonymous}::attemptToMoveSourceDbFiles(const rstudio::core::FilePath&, const rstudio::core::FilePath&) src/cpp/session/SessionSourceDatabaseSupervisor.cpp:215
Oct 15 09:07:14 vsci01-dev rsession-USER[260648]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:754
Oct 15 09:07:25 vsci01-dev rsession-USER[260669]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:07:26 vsci01-dev rsession-USER[260669]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:07:26 vsci01-dev rsession-USER[260669]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:754
Oct 15 09:07:37 vsci01-dev rsession-USER[260669]: ERROR system error 2 (No such file or directory); OCCURRED AT rstudio::core::Error rstudio::core::FilePath::getChildren(std::vector<rstudio::core::FilePath>&) const src/cpp/shared_core/FilePath.cpp:933; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::removeSessionDir(const rstudio::core::FilePath&) src/cpp/session/SessionSourceDatabaseSupervisor.cpp:150
Oct 15 09:08:33 vsci01-dev rsession-USER[261028]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/Git/tools/attribution-september-2024/.Rproj.user/63FFCF96/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:08:33 vsci01-dev rsession-USER[261028]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/Git/tools/attribution-september-2024/.Rproj.user/63FFCF96/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:08:34 vsci01-dev rsession-USER[261028]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:754
Oct 15 09:09:40 vsci01-dev rsession-USER[261028]: WARNING Private command disabled, unsupported $HISTCONTROL: [erasedups:ignoreboth]; LOGGED FROM: bool rstudio::core::terminal::PrivateCommand::output(const string&) src/cpp/core/terminal/PrivateCommand.cpp:265
Oct 15 09:09:54 vsci01-dev rsession-USER[261028]: ERROR system error 2 (No such file or directory); OCCURRED AT rstudio::core::Error rstudio::core::FilePath::getChildren(std::vector<rstudio::core::FilePath>&) const src/cpp/shared_core/FilePath.cpp:933; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::removeSessionDir(const rstudio::core::FilePath&) src/cpp/session/SessionSourceDatabaseSupervisor.cpp:150
Oct 15 09:10:26 vsci01-dev rsession-USER[261714]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:10:26 vsci01-dev rsession-USER[261714]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:10:52 vsci01-dev rsession-USER[261736]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:10:52 vsci01-dev rsession-USER[261736]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-1c39a6ab/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:10:53 vsci01-dev rsession-USER[261736]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:754
Oct 15 09:12:03 vsci01-dev rsession-USER[261736]: ERROR system error 2 (No such file or directory); OCCURRED AT rstudio::core::Error rstudio::core::FilePath::getChildren(std::vector<rstudio::core::FilePath>&) const src/cpp/shared_core/FilePath.cpp:933; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::removeSessionDir(const rstudio::core::FilePath&) src/cpp/session/SessionSourceDatabaseSupervisor.cpp:150
Oct 15 09:21:01 vsci01-dev rsession-USER[262237]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-5f8fc5f7/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:21:01 vsci01-dev rsession-USER[262237]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-5f8fc5f7/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:21:02 vsci01-dev rsession-USER[262237]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:754
Oct 15 09:21:13 vsci01-dev rsession-USER[262260]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-5f8fc5f7/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::attachToSourceDatabase() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:422
Oct 15 09:21:13 vsci01-dev rsession-USER[262260]: ERROR system error 37 (No locks available) [lock-file: /export/home/DEPARTMENT/USER/.local/share/rstudio/sources/session-5f8fc5f7/lock_file]; OCCURRED AT virtual rstudio::core::Error rstudio::core::LinkBasedFileLock::acquire(const rstudio::core::FilePath&) src/cpp/core/file_lock/LinkBasedFileLock.cpp:532; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::{anonymous}::createSessionDir() src/cpp/session/SessionSourceDatabaseSupervisor.cpp:247
Oct 15 09:21:14 vsci01-dev rsession-USER[262260]: ERROR The previous R session terminated abnormally; LOGGED FROM: rstudio::core::Error {anonymous}::rInit(const rstudio::r::session::RInitInfo&) src/cpp/session/SessionMain.cpp:754
``` 

### Describe the behavior you expected
Logging in & working with projects without error messages. 

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by krauthex** [comment]
My guess is that the issue lies somewhere in the NFS home setup, but that is unfortunately out of my hands (different department) and I don't have a good idea on where to begin looking .. :-/

**Comment by krauthex** [comment]
Ok, we did a a lot more digging since I posted this issue, and we found the reason for the problem (we think): we found a few old session folders under `/export/home/DEPARTMENT/USER/.local/share/rstudio/sources` (`/export/home` is the base path of the NFS global home) and RStudio tried to load some of them but they didn't have any content so RStudio complained about them. No idea where they came from, but we cleaned them up (also deleted everything in `~/.local/share/rstudio/session`), and now it appears to work flawlessly. 

Very strange, but I'm glad we apparently found the issue. Maybe this helps someone in the future. :-)
